Update plate 6 `samples`

jenna-tomkinson opened the pull request:
# Update Plate 6 

In this PR, `samples` parameter was updated for both normalize and feature select for Plate 6 to select samples from the iNFixion cell line and WT and Null genotypes. 

Should we also update this for the traditional bulk method?

gwaybio commented on 3.processing_features/scripts/2.pycytominer_singlecell_pipelines.py:
```diff
@@ -141,13 +141,16 @@
     if plate == "Plate_4":
         samples = "Metadata_Concentration == 0.0 and (Metadata_genotype == 'Null' or Metadata_genotype == 'WT')"
 
+    # Only for Plate 6, we want to normalize to iNFixion institution and Null and WT cells
```
can you add a brief phrase answering the question "why"?

jenna-tomkinson replied:
Will do!

gwaybio commented on 0.download_data/metadata/platemap_NF1_plate6.csv:
```diff
@@ -1,4 +1,4 @@
-WellRow,WellCol,well_position,gene_name,genotype,Institution ,seed_density
+WellRow,WellCol,well_position,gene_name,genotype,Institution,seed_density
```
nice fix! Just checking that you reprocessed these data now after updating the platemap?

jenna-tomkinson replied:
Yup! I had to do that to even process the data because I couldn't access the column properly.

gwaybio commented on 3.processing_features/2.pycytominer_singlecell_pipelines.ipynb:
```diff
@@ -157,8 +157,8 @@
-      "Performing normalization for Plate_6 using samples parameter: all\n",
-      "Aggregated dataframe shape (60, 1128)\n"
+      "Performing normalization for Plate_6 using samples parameter: Metadata_Institution == 'iNFixion' and (Metadata_genotype == 'Null' or Metadata_genotype == 'WT')\n",
+      "Aggregated dataframe shape (20, 1138)\n"
```
Hmmm, i don't think this should have updated the shape... all this should do is to identify the transformation based on the samples parameter. The data shape should be the same... please double check!

jenna-tomkinson replied:
I checked and it looks like there could be a bug in the "samples" parameter. I am going to check internally with the team to see if there is a fix or if this is a bug. I will be blocked on this until then.

jenna-tomkinson replied:
Bug has been fixed and looks like output is corrected!

gwaybio reviewed (commented):
An important question! Also, what is the feature selected dataframe shape? (it would be helpful to know how many features were removed)

jenna-tomkinson commented:
@gwaybio I have updated the environments to account for the bug fix. Plate 6 single cells have now been processed correctly with the samples parameter.

I did add two small changes to this PR that you can address:

1. Small refactor to regular expression in CP pipeline to correctly extract plate metadata, which is the first part to solve issue #57. I will need to rerun Plate_6 with CP to properly extract the plate metadata later.
2. I noticed the traditional bulk pipeline did not remove the contaminated HET cells or use `samples` parameter, so I updated it and reprocessed plates 3, 3 prime, 4, 5, and 6 just to be accurate. Depending on manuscript, we might not need these but I wanted to make sure it was accurate.

NOTE: Due to trying to install the correct version of pycytominer in my main environment (held pycytominer, cytotable, and cellprofiler), it was unfortunately a mistake as the environment broke and can no longer be used (of course cause cellprofiler). I had to delete the environment and reinstall, just so CellProfiler could work. Depending on your opinion, we could reprocess all of the plates with the new environment, but that might not be necessary.

gwaybio commented on 3.processing_features/scripts/0.merge_sc_cytotable.py:
```diff
@@ -118,19 +123,40 @@
-    # Columns to move to the front
-    columns_to_move = ['Nuclei_Location_Center_X', 'Nuclei_Location_Center_Y', 'Cells_Location_Center_X', 'Cells_Location_Center_Y']
+    # Update Image_Metadata_Plate column if plate is Plate_6
+    # TODO: Remove this once the metadata is fixed in the CellProfiler pipeline
```
Remove todo?

jenna-tomkinson replied:
Not yet! I will remove it once the plate is reran and I will make a PR for it! :)

jenna-tomkinson commented:
@gwaybio Thank you for the review! I am merging now!
